From david-b@pacbell.net Thu Dec 22 17:09:44 2005
From: David Brownell <david-b@pacbell.net>
To: Greg KH <greg@kroah.com>
Subject: USB: ehci fix driver model wakeup flags
Date: Thu, 22 Dec 2005 17:05:18 -0800
Cc: "Rafael J. Wysocki" <rjw@sisk.pl>, <gcoady@gmail.com>, Andrew Morton <akpm@osdl.org>
Message-Id: <200512221705.18618.david-b@pacbell.net>

On some systems, EHCI seems to be getting IRQs too early during driver
setup ... before the root hub is allocated, in particular, making trouble
for any code chasing down root hub pointers!  In this case, it seems to
be safe to just ignore the root hub setting.  Thanks to Rafael J. Wysocki
for getting this properly tested.

Signed-off-by: David Brownell <dbrownell@users.sourceforge.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/host/ehci-hcd.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- gregkh-2.6.orig/drivers/usb/host/ehci-hcd.c
+++ gregkh-2.6/drivers/usb/host/ehci-hcd.c
@@ -624,7 +624,7 @@ static irqreturn_t ehci_irq (struct usb_
 	}
 
 	/* remote wakeup [4.3.1] */
-	if ((status & STS_PCD) && device_may_wakeup(&hcd->self.root_hub->dev)) {
+	if (status & STS_PCD) {
 		unsigned	i = HCS_N_PORTS (ehci->hcs_params);
 
 		/* resume root hub? */
